narwhal/types/src/metered_channel.rs

```diff
@@ -267,6 +281,29 @@ pub fn channel<T>(size: usize, gauge: &IntGauge) -> (Sender<T>, Receiver<T>) {
         Receiver {
             inner: receiver,
             gauge: gauge.clone(),
+            total: None,
+        },
+    )
+}
+
+#[track_caller]
```

Comment:
Why here? When can this panic?

Reply:
Very good question, I just copied this from the `channel` definition above actually.